mcs/class/WindowsBase/System.Windows/Size.cs

```diff
@@ -62,31 +63,52 @@ public static bool Equals (Size size1, Size size2)
 			return size1.Equals (size2);
 		}
 
-		public override int GetHashCode ()
+		public override int GetHashCode()
 		{
-			throw new NotImplementedException ();
+			unchecked
+			{
+				return (_width.GetHashCode() * 397) ^ _height.GetHashCode();
+			}
 		}
 
 		public static Size Parse (string source)
 		{
-			throw new NotImplementedException ();
+			if (source == null)
+				throw new ArgumentNullException("source");
+			Size value;
+			if (source.Trim() == "Empty")
+			{
+				value = Empty;
+			}
+			else
+			{
+				var parts = source.Split(',');
+				if (parts.Length != 2)
+					throw new FormatException(string.Format("Invalid size string format: {0}", source));
+				value = new Size(Convert.ToDouble(parts[0].Trim(), CultureInfo.InvariantCulture),
```

Comment:
I don't use reference source before writing. I just check it after marek ask me.